lib/soapy_cake/time_converter.rb

```diff
@@ -0,0 +1,35 @@
+module SoapyCake
+  class TimeConverter
+    def initialize(time_zone, time_offset = nil)
+      if time_offset
+        self.class.print_deprecation_warning
+
+        # Etc/GMT time zones have their sign reversed
+        time_zone = format('Etc/GMT%+d', -time_offset.to_i)
+      end
+
+      fail Error, 'Cake time zone missing' if time_zone.blank?
+      @zone = ActiveSupport::TimeZone[time_zone]
+    end
+
+    def to_cake(date)
```

Comment:
Good point, was just looking at this method in isolation and thought of the possibility that a date is given as a string without a timezone. But as you are convinced that this method will always be used with proper Dates (incl. timezone) from within SoapyCake it's probably ok to be silent about the 'oh oh string without timezone' case.